tests/lib/rules/hook-use-state.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * @fileoverview Ensure symmetric naming of setState hook value and setter variables
+ * @author Duncan Beevers
+ */
+
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../../../lib/rules/hook-use-state');
+const parsers = require('../../helpers/parsers');
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    sourceType: 'module'
+  }
+});
+
+const tests = {
+  valid: [
+    {
+      code: 'const [color, setColor] = useState()'
```

Comment:
none of these validate that `useState` is used inside a react component. to avoid false positives, the rule should be limited to our react component detection.

Reply:
It's not uncommon to use these hooks outside of React components; for example, when writing custom hooks.

Reply:
That's a fair point - but in that case, we need to validate that `useState` is *react's* `useState`.

Reply:
In the explicit `React.useState` case, do you feel it's important to verify that `React` is actually React (eg, is preceded by `import React from 'react';` or somesuch?

Reply:
Yes.